addons/sale/report/report_all_channels_sales.py

```diff
@@ -37,10 +36,10 @@ def _so(self):
                     so.user_id AS user_id,
                     pt.categ_id AS categ_id,
                     so.company_id AS company_id,
-                    sol.price_total / COALESCE(cr.rate, 1.0) AS price_total,
+                    sol.price_total / CASE COALESCE(so.currency_rate, 0) WHEN 0 THEN 1.0 ELSE so.currency_rate END AS price_total,
```

Comment:
If the so.currency_rate is NULL, then the coalesce will return 0 so we enter the WHEN and take the THEN value. Isn't it simpler to use `COALESCE(so.currency_rate, 1.0)` directly, or did I miss something ?

Reply:
(same for the other SQL report ;) )

Reply:
@jem-odoo if currency_rate is equal to 0, I would like to avoid a 0 division. I can change the COALESCE with 1.0, but the case currency_rate=0 will still be possible.

Reply:
Indeed ! COALESCE take the first not NULL value, and not the one not evaluated to False. My bad ^^"